lib/ExpensiMark.js

```diff
@@ -223,6 +228,11 @@ export default class ExpensiMark {
                 regex: /<br(?:"[^"]*"|'[^']*'|[^'"><])*>\n?/gi,
                 replacement: '\n',
             },
+            {
+                name: 'heading1',
+                regex: /\s*<(h1)(?:"[^"]*"|'[^']*'|[^'">])*>(.*?)<\/\1>(?![^<]*(<\/pre>|<\/code>))\s*/gi,
```

Comment:
👋
This has caused a regression in https://github.com/Expensify/App/issues/17998. 
We were removing all of the new line breaks before `<h1>` tag, (\s), but should remove whitespaces ([^\S\r\n]).
More details [here](https://github.com/Expensify/App/issues/17998#issuecomment-1524741367)